ambry-router/src/main/java/com.github.ambry.router/AdaptiveOperationTracker.java

```diff
@@ -84,6 +84,10 @@
       localDcResourceToHistogram = getResourceToLatencyMap(routerOperation, true);
       crossDcResourceToHistogram = getResourceToLatencyMap(routerOperation, false);
     }
+    if (parallelism > routerConfig.routerAdaptiveTrackerMaxInflightRequests) {
```

Comment:
I think it would be good to also enforce this in `RouterConfig` constructor so that startup will fail if it is set improperly instead of failing continuously at runtime.

Reply:
Good point. Will make the change.